src/main/java/core/basesyntax/service/impl/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+
+public class PurchaseOperation implements OperationHandler {
+    private final FruitStorageDao storageDao;
+
+    public PurchaseOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void doOperation(String fruitName, Integer quantity) {
+        Integer fruitQuantity = storageDao.getQuantity(fruitName);
+        if (fruitQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruitName + " in the storage");
+        }
+        Integer previousQuantity = storageDao.getQuantity(fruitName);
+        Integer newQuantity = previousQuantity - quantity;
```

Comment:
You are calling `storageDao.getQuantity(fruitName)` twice to get the current quantity. You can optimize this by using the `fruitQuantity` variable obtained in line 14 for both the check and the calculation of `newQuantity`.